### PR TITLE
Dependencies between configurations

### PR DESCRIPTION
**Pull request by Eldblom**
This addresses the following issues:
- When a predicate's root is not a pre-existing root in Sitecore a warning shown on the Unicorn page. On sync the items are deserialized fine.
- If a solution has multiple configurations and one configuration is relying on templates and root items from another configuration there needs to be a way to explicitly set up dependencies.

This branch adds the following:
- Fixed the root paths to check the parent item and other configurations
- Added the ability to have dependencies between configurations
- Added explicit dependencies added in Configuration
- Added implicit dependencies based on root path

### CODE REVIEW

File: src/Unicorn/Unicorn.csproj [hunk 1]
```diff
@@ -214,6 +218,9 @@
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
   <Import Project="$(SolutionDir)\.nuget\NuGet.targets" Condition="Exists('$(SolutionDir)\.nuget\NuGet.targets')" />
+  <PropertyGroup>
```
Comment (kamsar) [inline src/Unicorn/Unicorn.csproj]: This looks like some local dev code that sneaked into the PR?

File: src/Unicorn/ControlPanel/Controls/ConfigurationDetails.cs [hunk 2]
```diff
@@ -1,115 +1,121 @@
...
+				</div>");
+        }
+
+        private void RenderDependencies(HtmlTextWriter writer)
+        {
+            var dependencies = ControlPanelUtility.FindConfigurationsDependencies(_configuration).ToArray();
+            if (dependencies.Any())
+            {
+                writer.Write(@"<h3>Dependencies</h3>");
+                writer.Write($@"<p class=""help"">This configuration is dependent on items in other configurations. Please review the following before synchronizing:</p>");
```
Comment (kamsar) [inline src/Unicorn/ControlPanel/Controls/ConfigurationDetails.cs]: Use of 'dependent' (en-US) and 'dependant' (en-UK) is inconsistent in text and variables. I'm fine with either one.

File: src/Unicorn/ControlPanel/Controls/ConfigurationDetails.cs [hunk 3]
```diff
@@ -1,115 +1,121 @@
-﻿using System.Web.UI;
+﻿#region using
```
Comment (kamsar) [inline src/Unicorn/ControlPanel/Controls/ConfigurationDetails.cs]: Please remove the `#region` if you don't mind.

File: src/Unicorn.Tests/Configuration/XmlConfigurationTypeActivatorTests.cs [hunk 4]
```diff
@@ -23,7 +23,7 @@ public void Activator_ResolvesType_WithDependency()
 		[Fact]
 		public void Activator_ResolvesType_WithDependency_AndParameter()
 		{
-			var micro = new MicroConfiguration("Test", null);
+			var micro = new MicroConfiguration("Test", null, null);
```
Comment (kamsar) [inline src/Unicorn.Tests/Configuration/XmlConfigurationTypeActivatorTests.cs]: No tests? Tsk tsk. I'll give you a pass unless you want to write them :)

**Comment by kamsar** [comment]
I love this PR. This is an excellent feature. I've been hacking around not having dependencies by using config declaration order so far (I define my content and code configs in the same file so that's easy).

My only issues are stylistic - the comments above, and that it looks like the code is formatted with spaces and Unicorn uses tabs (this makes reading the diffs particularly hard). Fix that and I'll merge it right up :)

**Comment by Eldblom** [comment]
Good to hear.
The feature is a necessity for sitecore/habitat where all modules own their own items in Sitecore, but rely on other modules for templates or root folders.
Sorry for not writing tests - not my forte - but I'll definitely fix the formatting issues asap.

**Comment by Eldblom** [comment]
I think this latest commit should fix your comments. Please let me know if you have additional suggesting before merging.

**Comment by Eldblom** [comment]
Any news on this?

**Comment by kamsar** [comment]
I've done a partial merge of this and opened #95 because there are still some todo items left before it gets fully merged. Going to close this and take the discussion to #95.

**Pull request closed without merging**